cpp/arcticdb/async/async_store.hpp

```diff
@@ -22,227 +22,263 @@ namespace arcticdb::async {
         return WriteSegmentTask{library_}(std::move(encoded));
     }
 
-    entity::VariantKey write_sync(PartialKey pk, SegmentInMemory &&segment) override {
+    entity::VariantKey write_sync(PartialKey pk, SegmentInMemory&& segment) override
+    {
         return write_sync(pk.key_type, pk.version_id, pk.stream_id, pk.start_index, pk.end_index, std::move(segment));
     }
 
-    entity::VariantKey write_sync(
-            KeyType key_type,
-            const StreamId& stream_id,
-            SegmentInMemory &&segment) override {
+    entity::VariantKey write_sync(KeyType key_type, const StreamId& stream_id, SegmentInMemory&& segment) override
+    {
         util::check(is_ref_key_class(key_type), "Expected ref key type got  {}", key_type);
         auto encoded = EncodeRefTask{key_type, stream_id, std::move(segment), codec_}();
         return WriteSegmentTask{library_}(std::move(encoded));
     }
 
-    folly::Future<folly::Unit> write_compressed(storage::KeySegmentPair&& ks) override {
+    folly::Future<folly::Unit> write_compressed(storage::KeySegmentPair&& ks) override
+    {
         return async::submit_io_task(WriteCompressedTask{std::move(ks), library_});
     }
 
-    void write_compressed_sync(storage::KeySegmentPair&& ks) override {
+    void write_compressed_sync(storage::KeySegmentPair&& ks) override
+    {
         library_->write(Composite<storage::KeySegmentPair>{std::move(ks)});
     }
 
-    folly::Future<entity::VariantKey> update(const entity::VariantKey &key, SegmentInMemory &&segment, storage::UpdateOpts opts) override {
+    folly::Future<entity::VariantKey>
+    update(const entity::VariantKey& key, SegmentInMemory&& segment, storage::UpdateOpts opts) override
+    {
         auto stream_id = variant_key_id(key);
         util::check(segment.descriptor().id() == stream_id,
-                    "Descriptor id mismatch in variant key {} != {}",
-                    stream_id,
-                    segment.descriptor().id());
+            "Descriptor id mismatch in variant key {} != {}",
+            stream_id,
+            segment.descriptor().id());
 
-        return async::submit_cpu_task(EncodeSegmentTask{
-            key, std::move(segment), library_, codec_, size_t(0)
-        })
-        .via(&async::io_executor())
-        .thenValue(UpdateSegmentTask{library_, opts});
+        return async::submit_cpu_task(EncodeSegmentTask{key, std::move(segment), library_, codec_, size_t(0)})
+            .via(&async::io_executor())
+            .thenValue(UpdateSegmentTask{library_, opts});
     }
 
-    folly::Future<VariantKey> copy(KeyType key_type, const StreamId& stream_id, VersionId version_id, const VariantKey& source_key) override {
-        return async::submit_io_task(CopyCompressedTask<ClockType>{source_key, key_type, stream_id, version_id, library_});
+    folly::Future<VariantKey>
+    copy(KeyType key_type, const StreamId& stream_id, VersionId version_id, const VariantKey& source_key) override
```

Comment:
Is it possible to prefer wrapping the arguments first instead of the type?

Reply:
I don't think there's a direct option to this. But you can assign some penalty value, so it prefers putting the type on the same line.